model_analyzer/triton/server/server_docker.py

```diff
@@ -68,8 +68,11 @@ def __init__(self, image, config, gpus, log_path, mounts, labels, shm_size):
         assert self._server_config['model-repository'], \
             "Triton Server requires --model-repository argument to be set."
 
-        logger.info(f"Pulling docker image {self._tritonserver_image}")
-        self._docker_client.images.pull(self._tritonserver_image)
+        try:
+            self._docker_client.images.get(self._tritonserver_image)
+        except:
+            logger.info(f"Pulling docker image {self._tritonserver_image}")
+            self._docker_client.images.pull(self._tritonserver_image)
```

Comment:
Does there also need to be a `try` around the pull? What happens if the pull fails?

Reply:
It will fail -- as it should. At that point we don't have any image to use.  
I could put a better failure message there though. I'm not sure if nested `try` blocks is good practice though.